docs/learning/ui/framework/Frontstages.md

```diff
@@ -8,153 +8,136 @@ A **Frontstage** is a full-screen configuration designed to enable the user to a
-              <Widget id="BottomEnd2" label="End2" element={<h2>Bottom End2 widget</h2>} />,
-            ],
-          },
-        }}
-      />
-    }
-  />
+export class FrontstageUi2 {
+  private static _contentGroupProvider = new FrontstageUi2ContentGroupProvider();
+  private static showCornerButtons = true;
+
+  public static supplyAppData(_id: string, _applicationData?: any) {
+    return {
+      viewState: UiFramework.getDefaultViewState,
+      iModelConnection: UiFramework.getIModelConnection,
+    };
+  }
+
+  public static register() {
+    // set up custom corner button where we specify icon, label, and action
+    const cornerButton = FrontstageUi2.showCornerButtons ?
+      <BackstageAppButton key="ui2-backstage" label="Toggle Ui2 Backstage" icon={"icon-bentley-systems"}
+        execute={() => BackstageManager.getBackstageToggleCommand().execute()} /> : undefined;
+    const hideNavigationAid = !FrontstageUi2.showCornerButtons;
+    const setUpCustomToolGroups = true;
+    const applicationData = setUpCustomToolGroups ? {
+      defaultContentTools: {
+        vertical: {
+          selectElementGroupPriority: 100,
+          measureGroupPriority: 200,
+          selectionGroupPriority: 300,
+        },
+        horizontal: {
+          clearSelectionGroupPriority: 100,
+          overridesGroupPriority: 200,
+        },
+      },
+    } : undefined;
+
+    const ui2StageProps: StandardFrontstageProps = {
+      id: "Ui2",
+      version: 1.1,
+      contentGroupProps: FrontstageUi2._contentGroupProvider,
+      hideNavigationAid,
+      cornerButton,
+      usage: StageUsage.General,
+      applicationData,
+    };
+
+    ConfigurableUiManager.addFrontstageProvider(new StandardFrontstageProvider(ui2StageProps));
+    this.registerToolProviders();
+  }
+
+  private static registerToolProviders() {
+
+    // Provides standard tools for ToolWidget in ui2.0 stage
+    StandardContentToolsProvider.register("ui2-standardContentTools", {
+      horizontal: {
+        clearSelection: true,
+        clearDisplayOverrides: true,
+        hide: "group",
+        isolate: "group",
+        emphasize: "element",
+      },
+    }, (stageId: string, _stageUsage: string, _applicationData: any) => {
+      return stageId === "Ui2";
+    });
+
+    // Provides standard tools for NavigationWidget in ui2.0 stage
+    StandardNavigationToolsProvider.register("ui2-standardNavigationTools", undefined, (stageId: string, _stageUsage: string, _applicationData: any) => {
+      return stageId === "Ui2";
+    });
+
+    // Provides standard status fields for ui2.0 stage
+    StandardStatusbarItemsProvider.register("ui2-standardStatusItems", undefined, (stageId: string, _stageUsage: string, _applicationData: any) => {
+      return stageId === "Ui2";
+    });
+
+    // Provides example widgets ui2.0 stage
+    AppUi2StageItemsProvider.register(FrontstageUi2.showCornerButtons);
```

Comment:
This code snippet does not come with `AppUi2StageItemsProvider` object definition.

Reply:
Added. It's a bit long, but it shows how to add UI in every region.